packages/react-instantsearch-hooks/src/components/__tests__/InstantSearch.test.tsx

```diff
@@ -749,4 +749,55 @@ describe('InstantSearch', () => {
       expect(searchFunction2).toHaveBeenCalledTimes(5);
     });
   });
+
+  test('triggers no more than one search on unmount', async () => {
```

Comment:
The only triggered search request is on render, so when unmounting we should expect the number of calls not to grow.

```suggestion
  test('triggers no search on unmount', async () => {
```